pulp_ansible/app/tasks/utils.py

```diff
@@ -39,3 +43,57 @@ def filter_namespace(metadata, url):
         metadata["results"] = results
 
     return metadata
+
+
+def parse_collections_requirements_file(requirements_file_string):
+    """
+    Parses an Ansible requirement.yml file and returns all the collections defined in it.
+
+    The requirements file is in the form:
+        ---
+        collections:
+        - namespace.collection
+        - name: namespace.collection
+            version: version identifier, multiple identifiers are separated by ','
+            source: the URL or prededefined source name in ~/.ansible_galaxy
+                    to pull the collection from
+    :param requirements_file_string: The string of the requirements file.
+    :return: A list of tuples (name, version, source).
+    """
+    collection_info = []
+
+    if requirements_file_string:
+        try:
+            requirements = yaml.safe_load(requirements_file_string)
+        except YAMLError as err:
+            raise ValidationError(
+                _(
+                    "Failed to parse the collection requirements yml at '%s' "
```

Comment:
This may not i8ln correctly in all languages with positional parameter replacement. It's possible some languages switch around the structure in a way that would swap the variables being replaced. Using `.format(key=value)` would solve this by giving each param a name.